### PR TITLE
Issue 503 allow list with noassertion

### LINKED ISSUE #503: Allow List with NOASSERTION/ NONE for PackageLicenseInfoFromFiles, LicenseInfoInSnippet and LicenseInfoInFile
The fields `PackageLicenseInfoFromFiles` in `Package`, `LicenseInfoInSnippet` in `Snippet` and `LicenseInfoInFile` in `File` can be populated with values like `[MIT, NOASSERTION]`. See [this issue ](https://github.com/spdx/spdx-spec/issues/809) in the spec repo. The current refactored version of the tools doesn't allow this which is a bug.

### PR DESCRIPTION
**Pull request by meretp**
fixes #503

### CODE REVIEW

**Review by armintaenzertng — CHANGES_REQUESTED** [review]
Thanks for this!
I found a few places that still can be adapted. Also, we can simplify the `file_converter`, `package_converter` and `snippet_converter` files in the `jsonschema` package.

File: tests/spdx/parser/rdf/data/file_to_test_rdf_parser.rdf.xml [hunk 1]
```diff
@@ -138,48 +140,7 @@
     </spdx:Package>
     <spdx:Package rdf:about="https://some.namespace#SPDXRef-Package2">
         <spdx:name>packageName</spdx:name>
-        <spdx:releaseDate>2022-12-01T00:00:00Z</spdx:releaseDate>
-        <spdx:supplier>Person: supplierName (some@mail.com)</spdx:supplier>
```
[the remark this answers is not in the document]
Comment (meretp) [reply tests/spdx/parser/rdf/data/file_to_test_rdf_parser.rdf.xml:142]: We use the second package to test different values in `ExternalPackageRef` (see [test_package_parser.py](https://github.com/spdx/tools-python/blob/refactor-python-tools/tests/spdx/parser/rdf/test_package_parser.py)). While working on this I decided that we don't need all these other fields in the second package and deleted it. I admit that this is not really related to the issue...

File: tests/spdx/fixtures.py [hunk 2]
```diff
@@ -85,7 +87,7 @@ def package_fixture(spdx_id="SPDXRef-Package", name="packageName", download_loca
                     valid_until_date=datetime(2022, 12, 3)) -> Package:
     checksums = [checksum_fixture()] if checksums is None else checksums
     license_info_from_files = [get_spdx_licensing().parse("MIT"), get_spdx_licensing().parse(
-        "GPL-2.0")] if license_info_from_files is None else license_info_from_files
+        "GPL-2.0"), SpdxNoAssertion()] if license_info_from_files is None else license_info_from_files
```
[the remark this answers is not in the document]
Comment (meretp) [reply tests/spdx/fixtures.py:90]: done

File: src/spdx/validation/license_expression_validator.py [hunk 3]
```diff
@@ -19,12 +19,11 @@
 from spdx.validation.validation_message import ValidationMessage, ValidationContext, SpdxElementType
 
 
-def validate_license_expressions(license_expressions: Optional[
-    Union[List[LicenseExpression], SpdxNoAssertion, SpdxNone]], document: Document, parent_id: str) -> List[ValidationMessage]:
-    if license_expressions in [SpdxNoAssertion(), SpdxNone(), None]:
```
[the remark this answers is not in the document]
Comment (meretp) [reply src/spdx/validation/license_expression_validator.py:24]: There is no `None` case, only an empty list, I fixed the type hint.

File: src/spdx/parser/jsonlikedict/file_parser.py [hunk 4]
```diff
@@ -53,7 +54,7 @@ def parse_file(self, file_dict: Dict) -> Optional[File]:
             lambda x: parse_field_or_no_assertion_or_none(x, self.license_expression_parser.parse_license_expression))
 
         license_info_in_files: Optional[
-            Union[List[LicenseExpression], SpdxNoAssertion, SpdxNone]] = parse_field_or_log_error(
+            List[Union[LicenseExpression, SpdxNoAssertion, SpdxNone]]] = parse_field_or_log_error(
             logger, file_dict.get("licenseInfoInFiles"),
             lambda x: parse_field_or_no_assertion_or_none(x, self.license_expression_parser.parse_license_expressions))
```
[the remark this answers is not in the document]
Comment (meretp) [reply src/spdx/parser/jsonlikedict/file_parser.py]: done

File: src/spdx/parser/jsonlikedict/package_parser.py [hunk 5]
```diff
@@ -73,19 +73,14 @@ def parse_package(self, package_dict: Dict) -> Package:
         homepage: Optional[str] = package_dict.get("homepage")
         license_comments: Optional[str] = package_dict.get("licenseComments")
         license_concluded = parse_field_or_log_error(
-            logger, package_dict.get("licenseConcluded"),
-            lambda x: parse_field_or_no_assertion_or_none(x, self.license_expression_parser.parse_license_expression),
-            None)
+            logger, package_dict.get("licenseConcluded"), self.license_expression_parser.parse_license_expression, None)
```
Comment (armintaenzertng) [inline src/spdx/parser/jsonlikedict/package_parser.py:76]: The `None` value is default anyway and `license_concluded` has not type hint

**Review by armintaenzertng — APPROVED** [review]
Thanks, only a small remark.

**Pull request merged**